### PR TITLE
upgrade to react 15.2.1

### PR DESCRIPTION
**Pull request by TomGallon**
## Improvement for react 15

- [x] React 15.2.1
- [x] ReactDOM 15.2.1
- [x] :package: 
- [x] :lollipop: 
- [x] We also made some code improvement to remove new react 15 warnings (unknown props passed to html element attributes)

### CODE REVIEW

File: src/list/mixin/built-in-components.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ var React = require('react');
 var fielBehaviourMixin = require('../../common/mixin/field-component-behaviour');
 var assign = require('object-assign');
 var Field = require('../../common/field').component;
-var Text = require('../../common/display/text').component;
+var DisplayText = require('../../components/display/text');
```
Comment (Ephrame) [inline src/list/mixin/built-in-components.js]: Aaaaaaaaaaaaaaaaaaaaaaaaaaaah des var !!!

Comment (TomGallon) [inline src/list/mixin/built-in-components.js]: yes but that's the way it was coded... i kept it !

**Comment by Bernardstanislas** [comment]
13451219_1628829827439780_1628365825_n

**Comment by pierr** [comment]
:chicken:

**Comment by pierr** [comment]
:crown:

File: src/common/mixin/built-in-components.js [hunk 2]
```diff
@@ -214,11 +213,11 @@ You must migrate all the autocompleteFor to autocompleteSelectFor or autocomplet
...
             <Button
-                handleOnClick={handleOnClick}
-                icon='save'
-                label='button.save'
-                shape={null}
-                type='button'
+            handleOnClick={handleOnClick}
+            icon='save'
+            label='button.save'
+            shape={null}
```
Comment (pierr) [inline src/common/mixin/built-in-components.js]: :-1:

File: src/components/button/index.js [hunk 3]
```diff
@@ -133,16 +133,13 @@ class Button extends Component {
...
-            renderedClassName = className + ' ' + ::this._getComponentClassName();
-        }
-        else {
-            renderedClassName = ::this._getComponentClassName();
-        }
+        // attribute doc : https://developer.mozilla.org/fr/docs/Web/HTML/Element/Button
+        // be careful the way you declare your attribute names : https://developer.mozilla.org/fr/docs/Web/HTML/Element/Button
+        const {className, disabled, formNoValidate, icon, id, type, label, style, ...otherProps } = this.props;
+        const otherInputProps = { disabled, formNoValidate, style, type };
+        const renderedClassName = `${className ? className : ''} ${::this._getComponentClassName()}`.trim();
```
Comment (pierr) [inline src/components/button/index.js]: :whale:

File: src/components/input/select/index.js [hunk 4]
```diff
@@ -96,18 +96,17 @@ class Select extends Component {
...
+
     /**
     * @inheritdoc
     * @override
     */
     render() {
-        const {error, name, placeholder, style, value, values, disabled, onChange, ...otherProps} = this.props;
-        //const pattern = error ? 'hasError' : null; //add pattern to overide mdl error style when displaying an focus error.
-        const selectOtherProps = disabled ? {disabled: 'disabled', ...otherProps} : otherProps;
+        const { autoFocus, error, multiple, name, placeholder, style, value, values, disabled, onChange, size } = this.props;
```
Comment (pierr) [inline src/components/input/select/index.js]: you can remove this line

File: src/list/mixin/built-in-components.js (hunk 1 above)
Comment (pierr) [inline src/list/mixin/built-in-components.js]: you can use `const` on all of them :dollar:

File: src/search/facet-box/facet.js [hunk 5]
```diff
@@ -97,8 +99,9 @@ const Facet = {
             <div className='' data-focus='facet-data-list'>
                 <ul>
                     {facetValues.map( facetValue => {
+                        const itemKey = uniqueId('facet-item');
                         return (
-                            <li key={facetValue.label}>
+                            <li key={itemKey}>
```
Comment (pierr) [inline src/search/facet-box/facet.js]: you can use `uniqueId('facet-item')`

File: src/common/mixin/built-in-components.js (hunk 2 above)
Comment (TomGallon) [inline src/common/mixin/built-in-components.js]: we keep it like this

File: src/components/input/select/index.js (hunk 4 above)
Comment (TomGallon) [inline src/components/input/select/index.js]: nope

File: src/list/mixin/built-in-components.js (hunk 1 above)
Comment (TomGallon) [inline src/list/mixin/built-in-components.js]: we ll do it when we'll publish lib in version 2

File: src/components/button/index.js (hunk 3 above)
Comment (TomGallon) [inline src/components/button/index.js]: :helicopter:

**Comment by Ephrame** [comment]
:crown:

**Pull request merged**